sample-status-updater-application/src/main/java/life/qbic/samplestatusupdater/app/SampleUpdater.java

```diff
@@ -72,18 +74,30 @@ public void execute() {
         SampleTrackingService sampleTrackingService = new SampleTrackingServiceConnector(serviceList.get(0), credentials);
 
         OpenBisSearchService searchService = new SampleSearchConnector(session.getApi(), session.getToken());
-        Date lastSearchDate = parseDateFromStringWithPattern((String) properties.get("lastSearchDate"), "yyyy-MM-dd");
+        Instant lastSearchDate = null;
+        try{
+            lastSearchDate = Instant.parse((String) properties.get("lastSearchDate"));
+        } catch (Exception e) {
+            LOG.error("Could not parse last search date " + properties.get("lastSearchDate"));
```

Comment:
You could extract `properties.get("lastSearchDate")` into a local variable before the try-catch block. This way the code that is expected to potentially fail is reduced to `Instant.parse`.

Reply:
I argue the other way ;) I can capture both

Reply:
In case `properties.get("lastSearchDate")` fails, your catch block will fail as well 😉